ntoskrnl/ke/amd64/trap.S

```diff
@@ -461,8 +461,11 @@ IntsDisabled:
 
 PageFaultError:
 
-    /* Set parameter 1 to error code */
+    /* Set parameter 1 to write/execute flag.
+       See https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-exception_record */
     mov r9d, [rbp + KTRAP_FRAME_ErrorCode]
```

Comment:
Can you tell me in comment what the format of this error code initially stored in r9d is? (and how we come up with the correct flags you want afterwards with the bit flagging done below)

Reply:
The CPU pushes the fault code: http://wiki.osdev.org/Exceptions#Page_Fault
0x02 (bit 1) = Write, 0x10 (bit 4) = Instruction Fetch, mask them and shift them right by 1 and you get the flags for the exception.